File: 812-RotateString/812-RotateString.cpp
// Last updated: 7/9/2026, 9:48:01 AM
class Solution {
public:
    bool rotateString(string s, string goal) {
        if(s.length()!=goal.length()){
            return false;
        }
        string doubled= s+s;
        return doubled.contains(goal);
        
    }
};

File: 812-RotateString/812-RotateString.java
// Last updated: 7/9/2026, 9:47:35 AM
class Solution {
    public boolean rotateString(String s, String goal) {
          if(s.length()!=goal.length()){
            return false;
        }
        String str= s+s;
        return str.contains(goal);
        
    }
}